### ISSUE TITLE
[IV/TLAD/TBOGT] Night Shadows affect the water / Shadows popping up when changing the camera

### ISSUE THREAD

**Issue by anthoncast** [GTAmodding/GTAIV-Issues-List#114]
This bug report is a copypaste from GTAForums post I made. So it can be listed here and I hope it gets fixed in Fusion Fix.

Game version that I am playing is The Complete Edition (Steam latest version).

These are my specs:
Processor Intel(R) Core(TM) i5-10300H CPU @ 2.50GHz   2.50 GHz
RAM 8.00 GB
64-bit operating system, x64-based processor
NVIDIA GTX 1650
Windows 11 as my OS.

Important to mention that I have these arguments in the commandline:

-nomemrestrict -availablevidmem 4096

Using DXVK (but still these bugs happens without it) and I have locked this game to 30 frames BTW.

**First problem: Shadows popping up**

I am facing these shadow issues within the game. They act very weird, and as I recall, I didn't see this issue in PS3 / XBOX 360. Of course, some objects popping up in front of my face is pretty normal, but this is just happening with the shadows of the game environment and objects.

Now, this doesn't happen in 1040 version. This issue appears in 1070 as far I know, and it remained in 1080.
 
Video: https://www.youtube.com/watch?v=7rGNkm5Pqx0
 
 **The second problem: Night Shadows affecting the water (most noticeable if game is locked in 30/60 fps)**
For some odd reason the water flickers when the night shadows option is enabled, and it's worse when the fps are locked to 60 or 30. Somehow it doesn't flicker when it reaches 120 fps and above.

Video: https://youtu.be/SDjHoM9goJk

**Comment by RecklessGlue540** [comment]
> The second problem: Night Shadows affecting the water (most noticeable if game is locked in 30/60 fps)
For some odd reason the water flickers when the night shadows option is enabled, and it's worse when the fps are locked to 60 or 30. Somehow it doesn't flicker when it reaches 120 fps and above.
Video: https://youtu.be/SDjHoM9goJk

Actually, it does not depend on Night Shadows at all, the issue happens even with shadows completely off, and framerate does not matter.
I found out that the issue seems to be partially(?) affected by View / Detail Distance.

I also have a strong suspicion the issue can happen even on 360, but It needs very in depth testing.

**Comment by anthoncast** [comment]
Yeah, I discovered that Night Shadows is not responsible of this at all. This weird visual glitch occurs a lot depending on the camera constant angle change.
Night shadows option makes this visual glitch get closer to you, disabling it makes it go far away, although it's noticeable if you're on boat and going in high speeds, and same in cars if you go to Northwood I think.
In the intro sequence, the FPS kinda affect how visible this glitch is. As I said, the less FPS, the more the effect is noticeable, and if the games goes beyond 60 fps the water will look clean as it's meant to.

I kinda don't want to disable Night Shadows at all, because in the consoles versions there were some Night Shadows, especially in interiors such as the apartment, the Platypus, Mikhail's house, and it adds some depth to the shadows. Would be possible to make the game have the same system of shadows as in consoles? Like a port. XD

Oh and I have to mention this too. I have both versions and I tested. I have my XBOX One S, and there it runs the cut-scenes at 60 fps or something, but in comparison to PC, this doesn't occur at least in cut-scenes. I know, little details, but in gameplay I just hate it too.
